src/registrar/models/domain.py

```diff
@@ -279,55 +282,173 @@ def _create_host(self, host, addrs):
         example: [(ns1.okay.gov, 127.0.0.1, others ips)]"""
-        # TODO: ticket #848 finish this implementation
-        # must delete nameservers as well or update
-        # ip version checking may need to be added in a different ticket
+
+        # TODO-848: ip version checking may need to be added in a different ticket
 
         if len(hosts) > 13:
             raise ValueError(
                 "Too many hosts provided, you may not have more than 13 nameservers."
             )
         logger.info("Setting nameservers")
         logger.info(hosts)
-        for hostTuple in hosts:
-            host = hostTuple[0]
-            addrs = None
-            if len(hostTuple) > 1:
-                addrs = hostTuple[1:]
-            avail = self._check_host([host])
-            if avail:
-                createdCode = self._create_host(host=host, addrs=addrs)
-
-                # update the domain obj
-                if createdCode == ErrorCode.COMMAND_COMPLETED_SUCCESSFULLY:
-                    # add host to domain
-                    request = commands.UpdateDomain(
-                        name=self.name, add=[epp.HostObjSet([host])]
-                    )
 
-                    try:
-                        registry.send(request, cleaned=True)
-                    except RegistryError as e:
-                        logger.error(
-                            "Error adding nameserver, code was %s error was %s"
-                            % (e.code, e)
-                        )
+        # get the changes made by user and old nameserver values
+        (
+            deleted_values,
+            updated_values,
+            new_values,
+            oldNameservers,
+        ) = self.getNameserverChanges(hosts=hosts)
+
+        # TODO-848: Fix name here
+        successTotalNameservers = self._loop_through(
+            deleted_values, updated_values, new_values, oldNameservers
+        )
 
-        try:
-            self.ready()
-            self.save()
-        except Exception as err:
-            logger.info(
-                "nameserver setter checked for create state "
-                "and it did not succeed. Error: %s" % err
-            )
-        # TODO - handle removed nameservers here will need to change the state
-        #   then go back to DNS_NEEDED
+        # print("SUCCESSTOTALNAMESERVERS IS ")
+        # print(successTotalNameservers)
+
+        if successTotalNameservers < 2:
+            try:
+                print("DNS_NEEDED: We have less than 2 nameservers")
+                self.dns_needed()
+                self.save()
+            except Exception as err:
+                logger.info(
+                    "nameserver setter checked for dns_needed state "
+                    "and it did not succeed. Warning: %s" % err
+                )
+        elif successTotalNameservers >= 2 and successTotalNameservers <= 13:
+            try:
+                print(
```

Comment:
**comment** We don't usually keep bare `print` statements around. Either use `logger.info` if it's worth keeping or delete it before the PR.

Reply:
Apologies -- this is still a draft PR so thank you for bearing with me on the random print statements for local testing reasons and random helper function names! Will totally remove when the PR is more finalized (currently just a self reference to make sure I'm getting the right count) 😅